OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -78,18 +83,36 @@ protected virtual Activity InnerTick(Actor self, AttackBase attack)
 				return NextActivity;
 
 			var type = Target.Type;
-			if (!Target.IsValidFor(self) || type == TargetType.FrozenActor)
+
+			if (!Target.IsValidFor(self)) {
```

Comment:
Can we split these FrozenActor &rarr; Actor and Actor &rarr; FrozenActor checks to their own `ValidateFrozenActorTarget` function?